_posts/2019-08-20-service-objects.markdown

```diff
@@ -0,0 +1,97 @@
+---
+layout: post
+title:  "Service Objects: beyond fat models and skinny controllers"
+date: 2019-08-20 10:00:00
+categories: ["rails"]
+author: "cleiviane"
+---
+
+Service Objects is a controversial pattern for several different reasons: some developers like to use it, others like to use similar patterns and some think that they are just unnecessary due that fat models approach recommended to Rails apps.  
+
+Here at Ombulabs we like to use Service Objects whenever we can, we think it's a great way to keep our controllers skinny.
+
+In this post I would like to discuss my idea about service objects and why it's adopted by our team.
+
+<!--more-->
+
+## What is a Service Object?
+
+A Service is a stateless object that encapsulates a set of steps and usually has a single purpose. It's a great resource to decompose fat active record models while still maintaining the controllers thin, because we take away some validations that doesn't belong to a model or a controller.  
+
+Recently we added a new feature in our productized service FastRuby.io , to verify if the visitor is suitable for a discount or not, for that validation we need to check if the discount is inside a 24 hour window,  connect to an external API service to check if it's a valid discount code and then decide to show or hide the discounted price.
+
+This is not something we should do in a model. A model is a representation of a plain ruby object that can be instantiated. In this case we are going to instantiate any validator model, it's a good thing to avoid instantiate objects whenever it's possible.
```

Comment:
```suggestion
This is not something we should do in a model. A model is a representation of a plain ruby object that can be instantiated. In this case we are not going to instantiate any validator model, it's a good thing to avoid instantiating objects whenever it's possible.
```

Could you explain this a bit more? Why is it good to avoid instantiating objects?